backend/agents/grading_agent.py

```
@@ -90,16 +95,33 @@ def calculate_overall_score(scene_scores: str) -> str:
             except Exception as e:
                 return f"Error parsing scores: {str(e)}"
-            
-            if scores:
-                avg_score = sum(scores) / len(scores)
-                return f"Overall score: {avg_score:.1f} (average of {len(scores)} scenes)"
-            return "No valid scores to calculate"
         
-        return [analyze_user_responses, evaluate_learning_objectives, 
-                generate_detailed_feedback, calculate_overall_score]
+        @tool
+        def identify_learning_gaps(responses: str, expected_outcomes: str) -> str:
```

Ruff (0.13.1)

Unused function argument: `responses`

(ARG001)